sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -1253,20 +1214,53 @@ public virtual Response<ExtractKeyPhrasesResultCollection> ExtractKeyPhrasesBatc
         {
             Argument.AssertNotNullOrEmpty(documents, nameof(documents));
             options ??= new TextAnalyticsRequestOptions();
+            MultiLanguageBatchInput documentInputs = ConvertToMultiLanguageInputs(documents);
 
+            return ExtractKeyPhrasesBatch(documentInputs, options, cancellationToken);
+        }
+
+        private async Task<Response<ExtractKeyPhrasesResultCollection>> ExtractKeyPhrasesBatchAsync(MultiLanguageBatchInput batchInput, TextAnalyticsRequestOptions options, CancellationToken cancellationToken)
+        {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(ExtractKeyPhrasesBatch)}");
             scope.Start();
 
             try
             {
-                using Request request = CreateDocumentInputRequest(documents, options, KeyPhrasesRoute);
-                Response response = _pipeline.SendRequest(request, cancellationToken);
+                Response<KeyPhraseResult> result = await _serviceRestClient.KeyPhrasesAsync(batchInput, options.ModelVersion, options.IncludeStatistics, cancellationToken).ConfigureAwait(false);
+                var response = result.GetRawResponse();
 
                 switch (response.Status)
                 {
                     case 200:
-                        IDictionary<string, int> map = CreateIdToIndexMap(documents);
-                        ExtractKeyPhrasesResultCollection results = CreateKeyPhraseResponse(response, map);
+                        IDictionary<string, int> map = CreateIdToIndexMap(batchInput.Documents);
+                        ExtractKeyPhrasesResultCollection results = Transforms.ConvertToExtractKeyPhrasesResultCollection(result.Value, map);
+                        return Response.FromValue(results, response);
+                    default:
+                        throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response).ConfigureAwait(false);
+                }
```

Comment:
`KeyPhrasesAsync` already throws if the response status is not 200, right? This `switch` may be unnecessary.

Reply:
thank you so much for catching this! you are right. We no longer need this.
I modified it for KeyPhrases and created issue https://github.com/Azure/azure-sdk-for-net/issues/14442 to do the same in all other APIs